project/apps/questions-service/src/questions.service.ts

```diff
@@ -25,16 +27,36 @@ export class QuestionsService {
     this.supabase = createClient(supabaseUrl, supabaseKey);
   }
 
-  private handleError(operation: string, error: unknown): never {
-    this.logger.error(`Error during ${operation}:`, error);
-    throw error;
+  /**
+   * Handles errors by logging the error message and throwing an RpcException.
+   *
+   * @private
+   * @param {string} operation - The name of the operation where the error occurred.
+   * @param {any} error - The error object that was caught. This can be any type of error, including a NestJS HttpException.
```

Comment:
HttpException wrapped in an RpcException can be thrown here.

We use a NestJS HttpException wrapped in an RpcException
because
1. questions-service uses rpc to communicate with api-gateway
2. api-gateway will unwrap the RpcException, and throw the inner
   HttpException to the client.

The above workaround is suggested on
https://stackoverflow.com/questions/72152120/how-to-handle-rpcexception-in-nestjs

I haven't found an easier and more straightforward way for the client (web) to receive httpexceptions without having to do some crazy parsing in `rpc-exception.filter.ts`

Reply:
yea, was researching on this too previously cause all microservices communicates with RPC instead of HTTP. this prob the easiest way alr